app/src/main/java/com/example/shiv/mdbmemberquizminiproject1/GameScreenActivity.java

```diff
@@ -0,0 +1,268 @@
+        }
+        button3.setText(members[option2]);
+
+        option3 = getRandomMemberNumber();
+        while (option3 == option2 || option3 == option1) {
+            option3 = getRandomMemberNumber();
+        }
+        button4.setText(members[option3]);
+
+        int option4 = getRandomMemberNumber();
+        while (option4 == option2 || option4 == option1 || option4 == option3) {
+            option4 = getRandomMemberNumber();
+        }
+        button2.setText(members[option4]);
+
+        final int answer;
+        double x = Math.random();
+        if (x < 0.25) {
+            correctButton = button1;
+            answer = option1;
+        } else if (x >= 0.25 && x < 0.50)
+        {
+            correctButton = button2;
+            answer = option4;
+        } else if (x >= 0.50 && x < 0.75) {
+            correctButton = button3;
+            answer = option2;
+        } else {
+            correctButton = button4;
+            answer = option3;
+        }
+
+        String str = members[answer].replaceAll("\\s+","");
+        str = str.toLowerCase();
+        int id = getImageId(context, str);
+        imgbutton.setImageResource(id);
+        //img.setImageResource(id);
+
+        timer = new CountDownTimer(5000, 1000) {
+
+            public void onTick(long millisUntilFinished) {
+                countdown.setText("seconds remaining: " + millisUntilFinished / 1000);
+            }
+
+            public void onFinish() {
+                countdown.setText("Time is up");
+                reset();
+            }
+        };
+        timer.start();
+
+        imgbutton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(ContactsContract.Intents.Insert.ACTION);
+                intent.setType(ContactsContract.RawContacts.CONTENT_TYPE);
+                intent.putExtra(ContactsContract.Intents.Insert.NAME, members[answer]);
+                context.startActivity(intent);
+
+
+            }
+        });
+
+        cycleGame();
+    }
+
+    public static int getImageId(Context context, String imageName) {
+        return context.getResources().getIdentifier("drawable/" + imageName, null, context.getPackageName());
+    }
+
+    private int getRandomMemberNumber() {
+        return 0 + (int)(Math.random() * ((members.length - 1 + 0) + 1));
+    }
+
+
+
+    private void cycleGame() {
+
+        final Button button1 = (Button) findViewById(R.id.button2);
+        final Button button2 = (Button) findViewById(R.id.button3);
+        final Button button3 = (Button) findViewById(R.id.button4);
+        final Button button4 = (Button) findViewById(R.id.button5);
+        final ImageButton imgbutton = (ImageButton) findViewById(R.id.imageButton2);
+
+        Button endGame = (Button) findViewById(R.id.button6);
+
+
+
+
+
+        button1.setOnClickListener(new View.OnClickListener() {
```

Comment:
Would be much cleaner to define a private method that takes in a button and sets an OnClickListener to it, would reduce code length by 75%